code/API_definitions/qod-api.yaml

```diff
@@ -329,18 +414,221 @@ components:
       description: |
         Ports may be specified as a list of ranges or single ports.
     QosProfile:
+      description: |
+        Data type with attributes of a QosProfile
+      type: object
+      properties:
+        name:
+          $ref: "#/components/schemas/QosProfileName"
+          description: |
+            The name of the QoS profile.  This name is used to identify the profile when creating a session.
+            The name must be unique.
+        status:
+          $ref: "#/components/schemas/QosProfileStatusEnum"
+          description: |
+            The status of the QoS profile.  The status is set to ACTIVE when the profile is created.
+            The status is set to INACTIVE when the profile is deleted.
```

Comment:
Are we allowing deletion of a profile dynamically?
As `status` in not a required field in `QosProfile`, should we consider profiles with this field not populated by the provider as implied `active`? I wonder if it is a good idea to make `status` as a required field.

This would help developer GET a profile, check its `active` `status`  before using it rather than `active or empty` status. Thoughts?

Reply:
Good point.  Let's make this required.

Reply:
@jlurien I went with @sfnuser suggestion.

> This would help developer GET a profile, check its active status before using it rather than active or empty status. Thoughts?

The point to simplify code to allow for always checking the value.  Though good coding standards would always check for a value is not null before comparing it, even if it is required.

What about setting the default to be ACTIVE for this status and having it as a required field?